reconcile/terraform_resources.py

```diff
@@ -549,27 +549,49 @@ def init_working_dirs(
     return ts, working_dirs
 
 
+def filter_accounts_by_name(
+    accounts: Iterable[Mapping[str, Any]], filter: Iterable[str]
+) -> Collection[Mapping[str, Any]]:
+    return [ac for ac in accounts if ac["name"] in filter]
+
+
+def exclude_accounts_by_name(
+    accounts: Iterable[Mapping[str, Any]], filter: Iterable[str]
+) -> Collection[Mapping[str, Any]]:
+    return [ac for ac in accounts if ac["name"] not in filter]
+
+
+def validate_account_names(
+    accounts: Collection[Mapping[str, Any]], names: Collection[str]
+) -> None:
+    if len(accounts) != len(names):
+        missing_names = set(names) - {a["name"] for a in accounts}
+        raise ValueError(
+            f"Accounts {missing_names} were provided as arguments, but not found in app-interface. Check your input for typos or for missing AWS account definitions."
+        )
+
+
 def setup(
     dry_run: bool,
     print_to_file: str,
     thread_pool_size: int,
     internal: str,
     use_jump_host: bool,
-    account_names: Optional[Collection[str]],
+    include_accounts: Optional[Collection[str]],
+    exclude_accounts: Optional[Collection[str]],
 ) -> tuple[ResourceInventory, OC_Map, Terraform, ExternalResourceSpecInventory]:
     accounts = queries.get_aws_accounts(terraform_state=True)
-    if account_names:
-        accounts = [n for n in accounts if n["name"] in account_names]
-        if len(accounts) != len(account_names):
-            # Some of the passed account names don't exist in app-interface
-            acc_names = tuple(
-                a
-                for a in account_names
-                if a not in tuple(account["name"] for account in accounts)
-            )
-            raise ValueError(
-                f"Accounts {acc_names} were provided as arguments, but not found in app-interface. Check your input for typos or for missing AWS account definitions."
-            )
+    if not include_accounts and exclude_accounts:
+        excluding = filter_accounts_by_name(accounts, exclude_accounts)
+        validate_account_names(excluding, exclude_accounts)
+        accounts = exclude_accounts_by_name(accounts, exclude_accounts)
+        if len(accounts) == 0:
+            raise ValueError("You have excluded all aws accounts, verify your input")
```

Comment:
You could also create domain specific exceptions here such as `NoAwsAccountSelectedException` or something like that. This will help in catching for that specific exception and also avoiding the need for string matching with generic exceptions like `ValueError` . You can evaluate the same for other exceptions thrown in this PR.